go/master/service.go

```diff
@@ -34,29 +34,30 @@ type Chunk struct {
 // Task is the basic unit of data instances assigned to trainers.
 type Task struct {
 	ID     int
+	Epoch  int
 	Chunks []Chunk
 }
 
 type taskEntry struct {
-	Epoch      int
 	NumTimeout int
 	Task       Task
+	NumFailed  int
```

Comment:
Maybe we can remove `NumTimeout`, we can treat timeout as failure. We can add NumTimeout back if we really need it. Let's keep the system simple.